### PR TITLE
fix prompt of latest version

### PR DESCRIPTION
**Pull request by amoskong**
### add special prompt for lack of latest_patch_version

`Unable to retrieve version information` is prompted if you run a RC release
until GA version is released. We also prompt same in some error cases.

### This patch added special prompt for this normal case.

fix DeprecationWarning of SafeConfigParser

DeprecationWarning: The SafeConfigParser class has been renamed to ConfigParser
in Python 3.2. This alias will be removed in future versions. Use ConfigParser
directly instead.
  config = configparser.SafeConfigParser()

/CC @amnonh

### CODE REVIEW

**Comment by amoskong** [comment]
ping @amnonh

File: dist/common/scripts/scylla-housekeeping [hunk 1]
```diff
@@ -138,12 +138,15 @@ def check_version(ar):
             params = params + "&rtype=" + repo_type
         versions = get_json_from_url(version_url + params)
         latest_version = versions["version"]
-        latest_patch_version = versions["latest_patch_version"]
+        latest_patch_version = versions.get("latest_patch_version")
```
Comment (amnonh) [inline dist/common/scripts/scylla-housekeeping]: Why 140 is ok, but 141 isn't?

File: dist/common/scripts/scylla-housekeeping [hunk 2]
```diff
@@ -138,12 +138,15 @@ def check_version(ar):
...
     except Exception:
         traceln("Unable to retrieve version information")
         return
 
-    if latest_patch_version != latest_version:
+    if latest_patch_version is None:
```
Comment (amnonh) [inline dist/common/scripts/scylla-housekeeping]: what scenario are you handling here?
Can you also add some general note to the PR to give it context?

File: dist/common/scripts/scylla-housekeeping [hunk 3]
```diff
@@ -181,7 +184,7 @@ if args.config != "":
     if not os.path.isfile(args.config):
         traceln("Config file ", args.config, " is missing, terminating")
         sys.exit(0)
-    config = configparser.SafeConfigParser()
+    config = configparser.ConfigParser()
```
Comment (amnonh) [inline dist/common/scripts/scylla-housekeeping]: how is that relates to the change?

File: dist/common/scripts/scylla-housekeeping (hunk 1 above)
Comment (amoskong) [reply dist/common/scripts/scylla-housekeeping]: When we use RC, only returned version, without 'latest_path_version'

File: dist/common/scripts/scylla-housekeeping (hunk 2 above)
Comment (amoskong) [reply dist/common/scripts/scylla-housekeeping]: Did you see the commitlog?

File: dist/common/scripts/scylla-housekeeping (hunk 3 above)
Comment (amoskong) [reply dist/common/scripts/scylla-housekeeping]: I change this by a single patch.

when I see DeprecationWarning &  message `Unable to retrieve version information` , it seems something is wrong.

**Comment by slivne** [comment]
@amnonh ping

**Comment by amoskong** [comment]
@amnonh any feedback?

Do you think it's not a real issue? and the warning/error won't make user confuse?

**Comment by slivne** [comment]
@amnonh ping

**Comment by slivne** [comment]
@amnonh ping

**Comment by amnonh** [comment]
I'm doing changes on the server side, that RC/master versions will not cause warning printing

**Comment by amoskong** [comment]
> I'm doing changes on the server side, that RC/master versions will not cause warning printing

Good, let's close this PR.

**Comment by yakirgb** [comment]
@amnonh  DeprecationWarning still exist:
```
Feb 03 16:25:36 r1.tab.com scylla-housekeeping[35228]: /opt/scylladb/scripts/libexec/scylla-housekeeping:195: DeprecationWarning: The SafeConfigParser class has been renamed to ConfigParser in Python 3.2. This alias will be removed in future versions. Use ConfigParser directly instead.
Feb 03 16:25:36 r1.tab.com scylla-housekeeping[35228]: config = configparser.SafeConfigParser()
```

**Pull request closed without merging**